modules/priceFloors.js

```diff
@@ -295,11 +295,30 @@ export function updateAdUnitsForAuction(adUnits, floorData, auctionId) {
   });
 }
 
+export function pickRandomModel(modelGroups) {
+  const weightSum = modelGroups.reduce((accum, model) => accum += model.modelWeight, 0);
+  // we loop through the models subtracting the current model weight from our random number
+  // once we are at or below zero, we return the associated model
+  let random = Math.floor(Math.random() * weightSum + 1)
+  for (let i = 0; i < modelGroups.length; i++) {
+    random -= modelGroups[i].modelWeight;
+    if (random <= 0) {
+      return modelGroups[i];
+    }
+  }
+};
```

Comment:
Here is a jsbin to test this function.
https://jsbin.com/qihiqojovu/edit?js,console

Weights do not need to add up to 100. Only shown in jsbin for clarity.